src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,263 @@
 package co.inventorsoft.academy.collections.model;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Function;
 
 public class Range<T> implements Set<T> {
 
+
+    private Node[] buckets;
+    private static final Integer INITIAL_CAPACITY = 30;
+    private int size;
+    Node<T>current=null;
+    Node<T> start=null;
+
+    public Range(final int capacity) {
+        this.buckets = new Node[capacity];
+    }
+
+    public Range() {
+        this(INITIAL_CAPACITY);
+    }
+
     public int size() {
-        return 0;
+        return size;
     }
 
     public boolean isEmpty() {
-        return false;
+        return size == 0;
     }
 
     public boolean contains(Object o) {
-        return false;
+        int index = o.hashCode() % buckets.length;
+        Node bucket = buckets[index];
+        boolean flag = false;
+
+        if (bucket == null) {
+            return flag;
+        }
+
+        if (bucket.data.equals(o)) {
+            flag = true;
+        }
+        return flag;
     }
 
     public Iterator<T> iterator() {
-        return null;
+
+        Iterator<T> iterator = new Iterator<T>() {
+
+            private int currentIndex = 0;
+
+            @Override
+            public boolean hasNext() {
+                return currentIndex < size;
+            }
+
+            @Override
+            public T next() {
+                Object tmp = start;
```

Comment:
Don't you have to go through all buckets?

Reply:
What do you mean?

Reply:
Could you rephrase your question?

Reply:
So, you have **private Node[] buckets;** do you go through all of them?

Reply:
Depends on situation.If I understood you - yes.